### PR TITLE
Add pugClassNotation 'attribute' option

### PR DESCRIPTION
**Pull request by h0merjam**
This is a partial fix for #167 covering classes, I think it _should_ be fairly straightforward to implement for IDs next, please let me know if this direction is right and I'll get on to it!

Thanks!

### CODE REVIEW

File: src/printer.ts [hunk 1]
```diff
@@ -773,6 +775,20 @@ export class PugPrinter {
 			if (token.val !== true) {
 				this.result += `=${token.val}`;
 			}
+		} else if (token.name === 'class' && this.options.pugClassNotation === 'attribute') {
+			const val: string = token.val.slice(1, -1).trim();
```
Comment (Shinigami92) [inline src/printer.ts:779]: **thought:** Havn't tested the code yet, and I'm currently just in GitHub Web UI as writing this review.
But this looks dangerous, cause the `token.val` could be not quoted and include js code 🤔, could it?

File: src/printer.ts [hunk 2]
```diff
@@ -838,6 +854,24 @@ export class PugPrinter {
 			this.result += this.indentString.repeat(this.indentLevel);
 		}
 		this.wrapAttributes = false;
+
+		if (this.classLiteralToAttribute.length) {
+			if (this.previousToken?.type === 'start-attributes') {
+				this.result += '(';
+			}
+
+			if (this.previousToken?.type === 'attribute') {
```
Comment (Shinigami92) [inline src/printer.ts:863]: **suggestion (performance):** Using at least an `else if` here could safe one check, cause when it was already a `start-attribute` it cannot be an `attribute` anymore

File: src/printer.ts (hunk 1 above)
Comment (Shinigami92) [reply src/printer.ts:779]: You should add a test case for this 🙂

**Review by Shinigami92 — COMMENTED** [review]
Code is looking good until now 👍 
You can proceed like this
Do you want to handle `id` in another PR? Or currently just solve it for `class`?

**Comment by h0merjam** [comment]
You know what, a separate PR would be better. It's classes I really need and I'm not sure I'll get on to IDs soon. Hopefully this helps anyone else wanting to have a go at it!

**Comment by Shinigami92** [comment]
Could you update https://github.com/prettier/plugin-pug/blob/main/docs/guide/pug-specific-options.md
and add a test case that contains class with js code

e.g.

```pug
- var clazz = 'ma-2 px-1'
div(class=clazz)
```

**Comment by Shinigami92** [comment]
Ah, these whitespaces are necessary cause otherwise VuePress will render these in one line instead of adding a newline

**Comment by h0merjam** [comment]
Oh no! I have an overzealous markdown formatter I'm afraid. Whitespace is restored now...

File: src/printer.ts [hunk 3]
```diff
@@ -773,6 +775,20 @@ export class PugPrinter {
 			if (token.val !== true) {
 				this.result += `=${token.val}`;
 			}
+		} else if (token.name === 'class' && this.options.pugClassNotation === 'attribute') {
+			const val: string = isQuoted(token.val) ? token.val.slice(1, -1).trim() : token.val;
+			const classes: string[] = val.split(/\s+/);
```
Comment (Shinigami92) [inline src/printer.ts:780]: **thought:** I'm a little bit afraid of this line and don't know what it's doing if the class contains js code with spaces 🤔  
But I think I will merge it for now

**Review by Shinigami92 — APPROVED** [review]
Cause of work for my employer, I hope I can setup and bundle everything together this evening or over the weekend

**Pull request merged**